scripts/run-integration-tests.sh

```diff
@@ -8,36 +8,40 @@
 # script, they are treated as the directories containing the tests to run.
 # When no arguments are provided, all integration tests are executed.
 
-set -e -u
+set -eu
 
 fabric_dir="$(cd "$(dirname "$0")/.." && pwd)"
-
 cd "$fabric_dir"
 
-dirs=()
-if [ "${#}" -eq 0 ]; then
-  specs=()
-  specs=("$(grep -Ril --exclude-dir=vendor --exclude-dir=scripts "RunSpecs" . | grep integration)")
-  for spec in ${specs[*]}; do
-    dirs+=("$(dirname "${spec}")")
-  done
+declare -a tempTestDirs
+if [[ "${#}" -eq 0 ]]; then
+  tempTestDirs=($(go list -f '{{.Dir}}' ./... | grep -E '/integration($|/)' | sed "s,${fabric_dir},.,g"))
 else
-  dirs=("$@")
+  tempTestDirs=("$@")
 fi
 
+declare -a testDirs
+for dir in ${tempTestDirs[@]}; do
+    listInside=$(go list -f '{{.TestGoFiles}}' ${dir})
+    listOutside=$(go list -f '{{.XTestGoFiles}}' ${dir})
+    if [[ "${listInside}" != "[]" || "${listOutside}" != "[]" ]]; then
+        testDirs+=(${dir})
+    fi
+done
+
```

Comment:
I think you're making more difficult than it needs to be. `go list` is your friend.

This gives you all of the packages with integration tests in them.
```
go list -f '{{if or (len .TestGoFiles | ne 0) (len .XTestGoFiles | ne 0)}}{{println .ImportPath}}{{end}}' ./... | grep integration
```

If you need the directories (for ginkgo), substitute `{{.Dir}}` for `{{.ImportPath}}`. if you need a relative path, you'll have to munge that by piping it through `sed "s,${fabric_dir},.,g"`.

Reply:
This is exactly what I was looking for actually, I couldn't figure out how to do conditionals in the format string for `go list`

THANK YOU!